src/Pheanstalk.php

```diff
@@ -383,7 +383,7 @@ private function reconnect()
     {
         $this->connection->disconnect();
 
-        if ($this->using != PheanstalkInterface::DEFAULT_TUBE) {
+        if ($this->using !== null && $this->using != PheanstalkInterface::DEFAULT_TUBE) {
```

Comment:
Hmm, this happens only if an exception is thrown during the `useTube` command which is called after a reconnect.
Instead it makes more sense to just dispatch the command remove the hacky state chage.